spec/support/time_matcher.rb

```diff
@@ -11,7 +11,11 @@
 end
 
 RSpec::Matchers.define :eq_time do |expected_time|
-  match { |actual_time| (actual_time - expected_time).abs < 0.001 }
+  match do |actual_time|
+    return true if expected_time.nil? && actual_time.nil?
```

Comment:
I'd return a custom error (if possible 🤔) or fail when both are nil.

When reading a spec I'd rather see an explicit `expect(a.updated_at).to be_nil` rather than e.g. `expect(a.updated_at).to eq_time(b.updated_at)` (and not knowing that both are actually nil)

Reply:
Hm, that would be a surprising behavior to me 🤔. Note that this behavior is quite idiomatic, for example `be` and `eq` matchers do the same in case of nils:

```ruby
  foo = nil
  bar = nil
  expect(foo).to be(bar) # OK
  expect(foo).to eq(bar) # OK
```

> I'd rather see an explicit `expect(a.updated_at).to be_nil`

Yeah, that would definitely be better in some cases, but note that you still can do that after this change, at the same time there are cases when comparing two dates is simpler and more straightforward.

For example, when you test a controller and compare returned json results with a model loaded from db. Or when testing data exports. I ran into this issue when testing an export, I do this:

```ruby
    expect(data_row[7]).to eq_time(message.created_at)
    expect(data_row[8]).to eq_time(message.updated_at)
    expect(data_row[9]).to eq_time(message.deleted_at)
```

`deleted_at` may be `nil` if the message is not deleted, and I can test that message in exported data have the same  `deleted_at` attribute as the message in db in one line, otherwise I need to write an if-else here.